specification/connector/_types/Connector.ts

```diff
@@ -0,0 +1,231 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { Id } from '@_types/common'
+import { Dictionary } from '@spec_utils/Dictionary'
+import { UserDefinedValue } from '@spec_utils/UserDefinedValue'
+
+interface SelectOption {
+  label: string
+  value: string
+}
+
+interface Dependency {
+  field: string
+  value: string | number | boolean | null
+}
+
+enum DisplayType {
+  TEXTBOX = 'textbox',
+  TEXTAREA = 'textarea',
+  NUMERIC = 'numeric',
+  TOGGLE = 'toggle',
+  DROPDOWN = 'dropdown'
+}
+
+enum ConnectorFieldType {
+  STRING = 'str',
+  INTEGER = 'int',
+  LIST = 'list',
+  BOOLEAN = 'bool'
+}
+
+interface Validation {
+  constraint: string | number
+  type: string
+}
+
+export interface ConnectorConfigProperties {
+  category?: string
+  default_value: string | number | boolean | null
+  depends_on: Dependency[]
+  display: DisplayType
+  label: string
+  options: SelectOption[]
+  order?: number | null
```

Comment:
I revised the types and I think for majority of cases we can safely mark fields as optional (instead of making them nullable). The only exception are errors that are (string | null), I can think of some situations where we want to mark error as null (when connector gets back to a healthy state)

Reply:
Ok. So we have to implement https://github.com/elastic/elasticsearch-specification/issues/2049 that clearly distinguishes "null as optional" and "null as an actual value" in the spec. Adding that on my immediate todo list.

Reply:
Does it imply that this PR needs to wait until #2049 is implemented? We are not in a rush, just want to clarify

Reply:
Yes, because currently `null` is considered as the equivalent of "missing" in languages where objects are represented as data structures and not dictionaries (i.e. all strictly typed languages). #2049 is meant to capture "null is an actual value" in the API spec so that we can generate code that correctly handles this.